File: 14-week-11/01-days-2-to-3--depth-first-graph-traversal/ruby/graph_dfs.rb
def is_path(graph, vertex_a, vertex_b)
  # type your code in here
end

if __FILE__ == $PROGRAM_NAME
  graph = {
    jan: [:john, :jambaby],
    john: [:carl],
    jambaby: [],
    carl: [:jambaby],
    dave: []
  }
  
  puts "Expecting: true"
  puts is_path(graph, :jan, :carl)

  puts

  puts "Expecting: false"
  puts is_path(graph, :jan, :dave)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 14-week-11/01-days-2-to-3--depth-first-graph-traversal/javascript/graph_dfs.js
function isPath(graph, vertexA, vertexB) {
  // type your code here
}

if (require.main === module) {
  // add your own tests in here
  let graph = {
    jan: ["john", "jambaby"],
    john: ["carl"],
    jambaby: [],
    carl: ["jambaby"],
    dave: []
  };

  console.log("Expecting: true");
  console.log(isPath(graph, "jan", "carl"));

  console.log("");

  console.log("Expecting: false");
  console.log(isPath(graph, "jan", "dave"));
}

module.exports = isPath;

// Please add your pseudocode to this file
// And a written explanation of your solution
